File: mobilipro/lib/features/auth/domain/models/partner_profile_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'partner_profile_dto.g.dart';

@JsonSerializable()
class PartnerProfileDto {
  const PartnerProfileDto({
    required this.id,
    required this.name,
    this.email,
    this.phone,
    this.logoUrl,
    this.businessNumber,
    required this.enabled,
    this.registrationCode,
    this.approvalStatus,
    this.rejectionReason,
  });

  final int id;
  final String name;
  final String? email;
  final String? phone;
  final String? logoUrl;
  final String? businessNumber;
  final bool enabled;
  final String? registrationCode;
  final String? approvalStatus;
  final String? rejectionReason;

  factory PartnerProfileDto.fromJson(Map<String, dynamic> json) =>
      _$PartnerProfileDtoFromJson(json);

  Map<String, dynamic> toJson() => _$PartnerProfileDtoToJson(this);
}

File: backend/mobili-boot/src/main/java/com/mobili/backend/module/partner/dto/PartnerProfileDTO.java
package com.mobili.backend.module.partner.dto;

import lombok.Data;

@Data
public class PartnerProfileDTO {
    private Long id;
    private String name;
    private String email;
    private String phone;
    private String logoUrl;
    private String businessNumber;
    private boolean enabled;
    /** Code saisi par les gares à l’auto-inscription. */
    private String registrationCode;
    private String approvalStatus;
    private String rejectionReason;
    /** Pays de la société (voir Country) — filtre les villes proposées à la création de gare
     *  (StationService.resolveCity impose la même contrainte côté backend). */
    private Long countryId;
    private String countryName;
}